src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForStatement.cs

```diff
@@ -28,13 +28,10 @@ public override BoundNode VisitForStatement(BoundForStatement node)
 
             return RewriteForStatement(
                 node,
-                node.OuterLocals,
                 rewrittenInitializer,
                 rewrittenCondition,
                 rewrittenIncrement,
-                rewrittenBody,
-                node.BreakLabel,
-                node.ContinueLabel, node.HasErrors);
+                rewrittenBody);
         }
 
         private BoundStatement RewriteForStatement(
```

Comment:
Consider commenting or renaming to indicate this is the rewriting when no inner locals.